### PR TITLE
Harmonize the generation of SubscriptionWhere- & Where- input-types, so they reuse the same logic

### PR DESCRIPTION
**Pull request by Andy2003**
# Description

Harmonize the generation of SubscriptionWhere- & Where- input-types, so they reuse the same logic

This fixes 2 issues:

1. The correct usage of non-nullable array elements
2. the correct handling of interfaces with enabled experimental flag (`typename_IN`)

## Complexity

Complexity: Low

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by Andy2003** [comment]
The tests are currently failing b/c of a bug I fixed in #4509, wich should be merged 1st

File: packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts [hunk 1]
```diff
@@ -239,38 +226,30 @@ function makeRelationshipToInterfaceTypeWhereType({
...
-    if (interfaceImplementationsType && !experimental) {
-        interfaceFields["_on"] = interfaceImplementationsType;
-    }
-    if (!isEmptyObject(interfaceFields)) {
-        interfaceNodeType = schemaComposer.getOrCreateITC(
-            interfaceEntity.operations.subscriptionWhereInputTypeName,
-            (tc) => tc.addFields(interfaceFields)
-        );
-    }
+    // TODO do we really need different where-input types for subscriptions?
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/generate-subscription-where-type.ts]: I'm not sure this is a TODO

But to answer, yes, where input in subscriptions do not allow for relationship filtering as well as some features

File: packages/graphql/src/schema/generation/where-input.ts [hunk 2]
```diff
@@ -37,6 +37,8 @@ import { withAggregateInputType } from "./aggregate-types";
 import { augmentWhereInputTypeWithRelationshipFields } from "./augment-where-input";
 import { makeImplementationsWhereInput } from "./implementation-inputs";
 
+const isEmptyObject = (obj: Record<string, unknown>) => !Object.keys(obj).length;
```
Comment (angrykoala) [inline packages/graphql/src/schema/generation/where-input.ts]: purely for consistency, please change this to a `function` syntax

```suggestion
function isEmptyObject(obj: Record<string, unknown>): boolean {
    return !Object.keys(obj).length;
}
```

**Review by angrykoala — APPROVED** [review]

**Pull request merged**